plugins/module_utils/network/sonic/config/vlan_mapping/vlan_mapping.py

```diff
@@ -143,26 +150,26 @@ def _state_replaced(self, want, have, diff):
         :returns: the commands necessary to migrate the current configuration
                   to the desired configuration
         """
-        requests = []
         commands = []
-        commands_del = []
-
-        commands_del = self.get_replaced_delete_list(want, have)
-
-        if commands_del:
-            commands.extend(update_states(commands_del, "deleted"))
+        requests = []
+        replaced_config = self.get_replaced_config(want, have)
+
+        add_commands = []
+        if replaced_config:
+            del_requests = self.get_delete_vlan_mapping_requests(replaced_config, have,
+                                                                 is_delete_all=False,
+                                                                 state='replaced')
+            requests.extend(del_requests)
+            commands.extend(update_states(replaced_config, "deleted"))
+            add_commands = want
```

Comment:
setting add_commands to want can cause extra requests if there are other mappings passed into want where want and have match.

Reply:
Good finding! Thanks. As we redefine new "replaced" design, we will revisit "replaced" functions for all modules. So the change will be done later together with check and diff modes support.

Reply:
Okay with leaving this for a revisit!